src/cli/mod.rs

```diff
@@ -239,8 +308,10 @@ fn get_parsed<T>(
 fn get_config(config_path: &Option<PathBuf>) -> Result<HashMap<String, String>> {
     // First load from the config file
     let builder = match config_path {
-        Some(config_path) => Config::builder().add_source(File::from(config_path.to_owned())),
-        None => Config::builder(),
+        Some(config_path) if config_path.exists() => {
```

Comment:
Is this match guard necessary? I think `config-rs` knows how to handle this

Reply:
When I tested before with `cargo run -- --config foo` I got the following output:
``` 
Error: configuration file "foo" not found
```
If this is desired behavior I can revert the change

Reply:
Yeah that's desired behavior. If the user asks `lurk` to run with a certain config file we should not go on with a different configuration if the file doesn't exist. It's better to alert the user that what he wants to do is impossible.

Reply:
I haven't reviewed this in detail and am only responding to the comment immediately above.

I think it's probably useful that there be a default config location such that if the config file is present, it is used, and if not, it is skipped. It shouldn't be an error to not have provided a default config file, nor should special configuration be required to create one.

For reference see, 'all CLI programs that have optional default config files' (for example, shells, editors, daemons, etc. — really, what doesn't work like that?)

Hopefully what I wrote is consistent with what you're describing, but just in case not — that's my opinion. I wrote this because it sounds like you are saying that having advised `lurk` where to find the config file, the config file becomes non-optional. That doesn't sound ergonomic to me.

In my opinion, an empty config file should be a no-op; and a missing config file should be treated as empty.